File: 019_Dart/fizzbuzz.dart
// This is a comment.

bool isFizz(int i) {
  return (i % 3 == 0);
}

bool isBuzz(int i) {
  return (i % 5 == 0);
}

String fizzbuzz(int i) {
  if (isFizz(i) && isBuzz(i)) {
    return "FizzBuzz";
  } else if (isFizz(i)) {
    return "Fizz";
  } else if (isBuzz(i)) {
    return "Buzz";
  } else {
    return i.toString();
  }
}

void main() {
  for (var i = 1; i <= 100; ++i) {
    final result = fizzbuzz(i);
    print(result);
  }
}

File: 007_Rust/fizzbuzz.rs
// This is a comment.

fn fizzbuzz(i: i32) -> String {
    //defines lambda functions
    let is_fizz = |i| i % 3 == 0;
    let is_buzz = |i| i % 5 == 0;

    if (is_fizz(i) && is_buzz(i)) {
        "FizzBuzz".to_string()
    } else if (is_fizz(i)) {
        "Fizz".to_string()
    } else if (is_buzz(i)) {
        "Buzz".to_string()
    } else {
        i.to_string()
    }
}

fn main() {
    for i in 1..101 {
        let result = fizzbuzz(i);
        println!("{}", result);
    }
}
